### PR TITLE
Kelsonic 22181 searchable content field

### PR DESCRIPTION
**Pull request by kelsonic**
## Description
**Issue:** https://github.com/department-of-veterans-affairs/va.gov-team/issues/22181

This PR adds a `searchableContent` field for R&S articles to augment our search experience.

## Screenshots
image

image

image

## Acceptance criteria
- [x] adds a `searchableContent` field for R&S articles to augment our search experience

## Definition of done
- [x] Events are logged appropriately
- [x] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

File: src/applications/resources-and-support/hooks/useGetSearchResults.js [hunk 1]
```diff
@@ -65,18 +65,45 @@ export default function useGetSearchResults(articles, query, page) {
...
+          (keywordInstances, keyword) =>
+            keywordInstances +
+            article.searchableContent.toLowerCase()?.split(keyword)?.length -
+            1,
+          0,
+        ),
+
+        // Number of times the full query is found in the article's title.
+        wholePhraseMatchCountsTitle:
+          article.title.toLowerCase()?.split(query.toLowerCase())?.length - 1,
```
Comment (alexandec) [inline src/applications/resources-and-support/hooks/useGetSearchResults.js:87]: Potential for refactoring since this pattern is repeated a few times

**Review by alexandec — APPROVED** [review]

File: src/applications/resources-and-support/hooks/useGetSearchResults.js [hunk 2]
```diff
@@ -65,18 +65,45 @@ export default function useGetSearchResults(articles, query, page) {
...
+            ?.length - 1,
+
+        // Number of times the full query is found in the article's title, introText, searchableContent combined.
+        wholePhraseMatchCountsTotal:
           article.title.toLowerCase()?.split(query.toLowerCase())?.length -
           1 +
           (article.introText.toLowerCase()?.split(query.toLowerCase())?.length -
+            1) +
+          (article.searchableContent.toLowerCase()?.split(query.toLowerCase())
+            ?.length -
```
Comment (ncksllvn) [inline src/applications/resources-and-support/hooks/useGetSearchResults.js:106]: I'm wondering if we might be able to refactor to abstract some of this into a helper or store them in variables?

It looks like `wholePhraseMatchCountsTotal` is the sum of `wholePhraseMatchCountsTitle` + `wholePhraseMatchCountsIntroText` + `wholePhraseMatchCountsContent`, unless I'm reading it wrong.

It also seems like this line `article.SOME_PROPERTY.toLowerCase()?.split(query.toLowerCase())?.length - 1` could be a nice helper function. My coffee is still kicking but right now that line is hard for me to understand.

Comment (kelsonic) [reply src/applications/resources-and-support/hooks/useGetSearchResults.js]: Updated

File: src/applications/resources-and-support/hooks/useGetSearchResults.js (hunk 1 above)
Comment (kelsonic) [reply src/applications/resources-and-support/hooks/useGetSearchResults.js]: Updated

**Comment by kelsonic** [comment]
@ncksllvn more importantly I'm wondering what you think of the new `articles.json` size with this `searchableContent` field addition 🙂

**Review by alexandec — APPROVED** [review]

**Comment by ncksllvn** [comment]
@kelsonic Oh I missed that...so it's about 319% larger than the old. Its current file size is still fine but it means our scalability challenges will be hit sooner. I'm not sure what that limit will be...half a MB would be uncomfortable for me. A query-less search on `/find-forms/` yields roughly a 375 KB JSON file, so maybe that can be our ceiling.

From a quick check in the sitemap it looks there are about 67 Resources articles on the site right now reflected in that `articles.json`...So that's about 2 KB per article in the new version. That means we'll be at the size of Find Forms by 187 articles and at half a meg at 250.

I think this is fine to merge but we should get a gauge on how many more articles we intend to publish before Resources is out of beta. 

Long term, this is really beyond the limits of what I imagined a front-end search doing. IDK what team would do it, but we need to figure out a plan for a true API backing this. The easy searches like `resources/search?q=DSLogon` are the ones that would really benefit.

**Comment by kelsonic** [comment]
Sounds good @ncksllvn, I dropped your comment in the original ticket and tagged Jen Lee, if you feel good merging this could you hook me up with an ✅? I think we need a review from PW to merge

**Review by ncksllvn — APPROVED** [review]
Looks great!

**Pull request merged**